app/api/deps/auth.py

```
@@ -51,4 +55,24 @@ def get_rate_limiter() -> RateLimiter:
+    # 2. JWT Verification
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, settings.SECRET_KEY, algorithms=["HS256"])
+        user_id: str = payload.get("sub")
+        if user_id is None:
+            raise credentials_exception
+        return user_id
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)